internal/v1/handler_oscap.go

```diff
@@ -181,150 +182,439 @@ func (h *Handlers) GetOscapCustomizations(ctx echo.Context, distribution Distrib
+	var errs []BlueprintLintItem
+
+	if policyBP == nil {
+		return errs
 	}
 
-	for _, fsc := range bp.Customizations.GetFilesystems() {
-		if cust.Filesystem == nil || !slices.ContainsFunc(*cust.Filesystem, func(fs Filesystem) bool {
-			return fs.Mountpoint == fsc.Mountpoint
-		}) {
-			lintErrors = append(lintErrors, BlueprintLintItem{
-				Name:        "Compliance",
-				Description: fmt.Sprintf("mountpoint %s required by policy is not present", fsc.Mountpoint),
-			})
+	for _, pkg := range policyBP.GetPackagesEx(false) {
+		if currentCust == nil || currentCust.Packages == nil || !slices.Contains(*currentCust.Packages, pkg) {
+			if !fixup {
+				errs = append(errs, BlueprintLintItem{
+					Name:        "Compliance",
+					Description: fmt.Sprintf("package %s required by policy is not present", pkg),
+				})
+			}
 			if fixup {
-				cust.Filesystem = common.ToPtr(append(common.FromPtr(cust.Filesystem), Filesystem{
-					Mountpoint: fsc.Mountpoint,
-					MinSize:    fsc.MinSize,
-				}))
+				currentCust.Packages = common.ToPtr(append(common.FromPtr(currentCust.Packages), pkg))
 			}
 		}
 	}
+	return errs
+}
+
+// lintPackagesW validates packages that were required by policy in snapshot but no longer required
+func lintPackagesW(policyBP *blueprint.Blueprint, snapshotCust *Customizations) []BlueprintLintItem {
+	var warns []BlueprintLintItem
 
-	if services := bp.Customizations.GetServices(); services != nil {
-		for _, e := range services.Enabled {
-			if cust.Services == nil || cust.Services.Enabled == nil || !slices.Contains(*cust.Services.Enabled, e) {
-				lintErrors = append(lintErrors, BlueprintLintItem{
+	// Check for obsolete packages from snapshot - even if policyBP is nil,
+	// as the policy might have changed to one without package requirements
+	if snapshotCust != nil {
+		for _, pkg := range common.FromPtr(snapshotCust.Packages) {
+			if policyBP == nil || !slices.Contains(policyBP.GetPackagesEx(false), pkg) {
+				warns = append(warns, BlueprintLintItem{
 					Name:        "Compliance",
-					Description: fmt.Sprintf("service %s required as enabled by policy is not present", e),
+					Description: fmt.Sprintf("package %s is no longer required by policy", pkg),
 				})
+			}
+		}
+	}
+	return warns
+}
+
+// lintFilesystemsE validates filesystems required by current policy but missing from current blueprint
+func lintFilesystemsE(policyBP *blueprint.Blueprint, currentCust *Customizations, fixup bool) []BlueprintLintItem {
+	var errs []BlueprintLintItem
+
+	if policyBP == nil {
+		return errs
+	}
+
+	// Only check if policy actually defines filesystem requirements
+	if policyBP.Customizations != nil {
+		for _, fsc := range policyBP.Customizations.GetFilesystems() {
+			if currentCust == nil || currentCust.Filesystem == nil || !slices.ContainsFunc(*currentCust.Filesystem, func(fs Filesystem) bool { return fs.Mountpoint == fsc.Mountpoint }) {
+				if !fixup {
```

Comment:
Why don't we always add the error as a blueprint lint item? Not sure why this is conditional on fixup.